### PR TITLE
Add early stopping to double ensemble model, add example

### PR DESCRIPTION
**Pull request by chenditc**
Existing double ensemble model does not implement capability to leverage validation set and do early stopping. Added such capability while remaining the default behavior unchanged.

## Description
Added early_stopping_round option to double ensemble model.

## Motivation and Context

https://github.com/microsoft/qlib/issues/1208

The fixed epoch for sub model might make them overfit or underfit, depends on the dataset. Using validation set for early stopping can make sub model more robust.

## How Has This Been Tested?

- [x] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [x] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test: 
image

2. Your own tests:
Run with early stopping:
image

Run with default params:
image

## Types of changes

- [ ] Fix bugs
- [x] Add new feature
- [ ] Update documentation

### CODE REVIEW

**Comment by you-n-g** [comment]
Please fix the CI error
https://github.com/microsoft/qlib/actions/runs/3553539133/jobs/5969103534#step:9:17
Thanks

**Comment by chenditc** [comment]
> Please fix the CI error https://github.com/microsoft/qlib/actions/runs/3553539133/jobs/5969103534#step:9:17 Thanks

Lint error fixed.

**Comment by you-n-g** [comment]
Thanks.
It looks great now!
